data/make/main.c

```diff
@@ -0,0 +1,52 @@
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+typedef uint64_t ibool_t;
+typedef int64_t iint_t;
+
+typedef struct {
+    /* inputs */
+    /* these are 32-bit file handles, but storing them as 64-bit in the struct makes it easier to poke from Haskell */
+    iint_t input_fd;
+    iint_t output_fd;
+    iint_t drop_fd;
+    iint_t chord_fd;
+
+    /* outputs */
+    const char *error;
+    iint_t fact_count;
+    iint_t entity_count;
+
+    /* stuff */
+    size_t facts_limit;   /* maximum number of facts per entity-attribute */
+    ibool_t has_drop; /* whether to write discarded facts to output or drop log */
+} psv_config_t;
```

Comment:
This data structure is duplicated from one of the `data/sea/*.h` files. Not sure of a way to de-dupe this that isn't overly complex :(